controllers/nodehealthcheck_controller_test.go

```diff
@@ -1959,6 +1959,101 @@ var _ = Describe("Node Health Check CR", func() {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(4))
+					}, "15s", "100ms").Should(Succeed())
+
+					// Verify Storm recovery is activated
+					// 7 total, 3 healthy, 4 unhealthy, 3 remediations (additional remediation not created because of min healthy constraint)
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(3))
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(4))
+						g.Expect(getRemediationsCount(underTest)).To(Equal(3))
+						g.Expect(utils.IsConditionTrue(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
+						g.Expect(underTest.Status.StormRecoveryStartTime).ToNot(BeNil())
+					}, "5s", "100ms").Should(Succeed())
+
+					// Phase 3: Recover one node - storm recovery remains active due to 2-second delay
+					By("recovering one node - storm recovery remains active due to 2-second delay")
+					mockNodeGettingHealthy("unhealthy-worker-node-1")
+
+					//wait for node to recover
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(4))
+					}, "5s", "100ms").Should(Succeed())
+					// 7 total, 4 healthy, 3 unhealthy, 2 remediations (one removed due to healthy node), 1 remediation is pending to be created (not created because of storm)
+					Consistently(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(4))
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(3))
+						g.Expect(getRemediationsCount(underTest)).To(Equal(2))
+						g.Expect(utils.IsConditionTrue(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
+					}, "2s", "100ms").Should(Succeed())
+					//expected termination time of the first storm
+					firstStormTerminationTime := underTest.Status.StormTerminationStartTime.Time.Add(underTest.Spec.StormTerminationDelay.Duration)
+					// Phase 4: wait for the delay to pass
+					time.Sleep(time.Millisecond * 1500)
+					//Expect Storm Recovery mode to end
+					// 7 total, 4 healthy, 3 unhealthy, 3 remediations (pending remediation created when storm is done)
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(utils.IsConditionSet(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
```

Comment:
We can use `IsConditionTrue`, no?
```suggestion
						g.Expect(utils.IsConditionTrue(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
```

Reply:
Thinking about it again, we can remove it entirely as the next line already does the "job" of verifying the status.

Reply:
>The condition is set
It equals false

What is the point of testing it twice?
We get the condition object twice based on the condition type, and if it isn't set, why do we need to check its value?